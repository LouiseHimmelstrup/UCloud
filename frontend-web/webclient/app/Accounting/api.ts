export interface DataPoint {
    label: string | null
}

export interface DataPoint1D<X = number> extends DataPoint {
    x: X
}

export interface DataPoint2D<X = number, Y = number> extends DataPoint1D<X> {
    y: Y
}

export interface DataPoint3D<X = number, Y = number, Z = number> extends DataPoint2D<X, Y> {
    z: Z
}

export interface Chart<Point extends DataPoint> {
    chartTypeHint?: string
    data: Point[]

    dataTitle?: string

    /**
     * An array of data types. Each element corresponds to a dimension. 
     * 
     * The data types are typically values from ChartDataTypes, but are allowed
     * to be of a different type.
     */
    dataTypes?: (string | null)[]
}

/**
 * Contains known data types. 
 * 
 * A data type is allowed to not be one of the following.
 */
export namespace ChartDataTypes {
    export const BYTES = "bytes"
    export const DURATION = "duration"
    export const DATE = "date"
    export const DATETIME = "datetime"
    export const NUMBER = "number";
}