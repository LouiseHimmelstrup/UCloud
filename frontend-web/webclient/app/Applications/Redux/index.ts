import * as ViewObject from "./ViewObject";
import * as ViewReducer from "./ViewReducer";
import * as ApplicationObject from "./ApplicationsObject";
import * as ApplicationReducer from "./ApplicationsReducer";

export type Reducers = ViewReducer.Reducer & ApplicationReducer.Reducer;
export type Objects = ViewObject.Wrapper & ApplicationObject.Wrapper;

export function init(): Objects {
    return { 
        ...ViewObject.init(),
        ...ApplicationObject.init()
    }
}

export const reducers: Reducers = {
    applicationView: ViewReducer.default,
    applicationsBrowse: ApplicationReducer.default
};