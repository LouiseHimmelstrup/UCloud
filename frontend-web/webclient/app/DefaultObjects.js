import {Cloud} from "../authentication/SDUCloudObject"
import {tusConfig} from "./Configurations";
import Uppy from "uppy";

export const DefaultStatus = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended.",
};

export const RightsMap = {
    NONE: 0,
    READ: 1,
    READ_WRITE: 2,
    OWN: 3,
};

export const AnalysesStatusMap = {
    "PENDING": 0,
    "IN PROGRESS": 1,
    "COMPLETED": 2,
};

export const RightsNameMap = {
    NONE: "None",
    READ: "Read",
    READ_WRITE: "Read/Write",
    OWN: "Own",
};

export const SensitivityLevel = {
    "OPEN_ACCESS": "Open Access",
    "CONFIDENTIAL": "Confidential",
    "SENSITIVE": "Sensitive",
};

export const SensitivityLevelMap = {
    "OPEN_ACCESS": 0,
    "CONFIDENTIAL": 1,
    "SENSITIVE": 2,
};

const initializeUppy = (restrictions) => 
    Uppy.Core({
        autoProceed: false,
        debug: false,
        restrictions: restrictions,
        meta: {
            sensitive: false,
        },
        onBeforeUpload: () => {
            return Cloud.receiveAccessTokenOrRefreshIt().then((data) => {
                tusConfig.headers["Authorization"] = `Bearer ${data}`;
            });
        }
    }).use(Uppy.Tus, tusConfig);

export const initObject = {
    files: {
        files: [],
        filesShown: 10,
        filesPerPage: 10,
        currentFilesPage: 0,
        loading: false,
        path: "",
        filesInfoPath: "",
        filesInfoFile: null,
        projects: []
    },
    uppy: {
        uppyFiles: initializeUppy({maxNumberOfFiles: false}),
        uppyFilesOpen: false,
        uppyRunApp: initializeUppy({ maxNumberOfFiles: 1}),
        uppyRunAppOpen: false
    },
    status: {
        status: DefaultStatus,
        title: ""
    }
};