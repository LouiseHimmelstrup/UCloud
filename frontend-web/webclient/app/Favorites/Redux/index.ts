import * as FavoritesObject from "./FavoriteObject";
import * as FavoritesReducer from "./FavoritesReducer";

export type Reducers = FavoritesReducer.Reducer;
export type Objects = FavoritesObject.Wrapper;

export function init(): Objects {
    return {
        ...FavoritesObject.init()
    }
}

export const reducers: Reducers = {
    favorites: FavoritesReducer.default
}