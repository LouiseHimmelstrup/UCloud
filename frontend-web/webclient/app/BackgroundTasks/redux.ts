import {Speed, TaskUpdate} from "BackgroundTasks/api";
import {Action} from "redux";
import {Dictionary} from "Types";
import {takeLast} from "Utilities/CollectionUtilities";

export interface TaskReduxState {
    tasks?: Dictionary<TaskUpdate>;
}

interface TaskUpdateAction extends Action<"TASK_UPDATE"> {
    update: TaskUpdate;
}

export function taskUpdateAction(update: TaskUpdate): TaskUpdateAction {
    return {
        type: "TASK_UPDATE",
        update
    };
}

type TypeActions = TaskUpdateAction;

export const reducer = (
    state: any = ({}),
    action: TypeActions
): TaskReduxState => {
    switch (action.type) {
        case "TASK_UPDATE": {
            const existingTask: TaskUpdate | undefined = state ? state[action.update.jobId] : undefined;
            if (!existingTask) {
                const tasks = {...state};
                tasks[action.update.jobId] = {
                    ...action.update,
                    speeds: insertTimestamps(action.update.speeds)
                };
                return tasks;
            } else {
                const currentMessage = existingTask.messageToAppend ? existingTask.messageToAppend : "";
                const messageToAdd = action.update.messageToAppend ? action.update.messageToAppend : "";
                const newMessage = currentMessage + messageToAdd;

                const newStatus = action.update.newStatus ? action.update.newStatus : existingTask.newStatus;
                const newTitle = action.update.newTitle ? action.update.newTitle : existingTask.newTitle;
                const newProgress = action.update.progress ? action.update.progress : existingTask.progress;
                const newSpeed = takeLast((existingTask.speeds || []).concat(action.update.speeds || []), 500);
                const newComplete = action.update.complete ? action.update.complete : existingTask.complete;

                const tasks = {...state};
                tasks[action.update.jobId] = {
                    ...existingTask,
                    messageToAppend: newMessage,
                    progress: newProgress,
                    speeds: insertTimestamps(newSpeed),
                    complete: newComplete,
                    newStatus,
                    newTitle
                };

                return tasks;
            }
        }
    }
    return state;
};

function insertTimestamps(speeds: Speed[]): Speed[] {
    return speeds.map(it => {
        if (it.clientTimestamp) {
            return it;
        } else {
            return {...it, clientTimestamp: Date.now()};
        }
    });
}
