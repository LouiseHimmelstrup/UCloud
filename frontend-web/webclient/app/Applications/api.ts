import {Client} from "@/Authentication/HttpClientInstance";
import {snackbarStore} from "@/Snackbar/SnackbarStore";
import {b64EncodeUnicode} from "@/Utilities/XHRUtils";
import {inSuccessRange} from "@/UtilityFunctions";
import {appLogoCache, toolLogoCache} from "@/Applications/AppToolLogo";
import {compute} from "@/UCloud";
import ApplicationSummaryWithFavorite = compute.ApplicationSummaryWithFavorite;
import {buildQueryString} from "@/Utilities/URIUtilities";

export interface ApplicationGroup {
    id: number,
    title: string,
    logo?: string,
    description?: string,
    application?: ApplicationSummaryWithFavorite
}

export interface CreateGroupRequest {
    title: string;
}

export interface CreateGroupResponse {}

export interface RetrieveGroupRequest {
    id?: string;
    name?: string;
}

export interface RetrieveGroupResponse {
    group: ApplicationGroup;
    applications: ApplicationSummaryWithFavorite[];
}


export interface SetGroupRequest {
    groupId: number,
    applicationName: string
}

export interface SetGroupResponse{}

export interface ListGroupsRequest {}

export interface UpdateGroupRequest {
    id: number,
    title: string,
    description?: string,
    logo?: Blob
}

export interface UpdateGroupResponse {}

export interface DeleteGroupRequest {
    id: number,
}

export interface DeleteGroupResponse {}

export function retrieveGroup(
    request: RetrieveGroupRequest 
): APICallParameters<RetrieveGroupRequest, RetrieveGroupResponse> {
    return {
        context: "",
        method: "GET",
        path: buildQueryString("/api/hpc/apps" + "/group", {id: request.id, name: request.name}),
        parameters: request,
        reloadId: Math.random(),
    }
}



export function createGroup(
    request: CreateGroupRequest 
): APICallParameters<CreateGroupRequest, CreateGroupResponse> {
    return {
        context: "",
        method: "POST",
        path: buildQueryString("/api/hpc/apps" + "/group", {}),
        parameters: request,
        reloadId: Math.random(),
        payload: request
    }
}

export function setGroup(
    request: SetGroupRequest 
): APICallParameters<SetGroupRequest, SetGroupResponse> {
    return {
        context: "",
        method: "POST",
        path: buildQueryString("/api/hpc/apps" + "/group/set", {}),
        parameters: request,
        reloadId: Math.random(),
        payload: request
    }
}

export function listGroups(
    request: ListGroupsRequest
): APICallParameters<ListGroupsRequest, ApplicationGroup[]> {
    return {
        context: "",
        method: "GET",
        path: buildQueryString("/api/hpc/apps" + "/groups", {}),
        parameters: request,
        reloadId: Math.random(),
    }
}

export function updateGroup(
    request: UpdateGroupRequest
): APICallParameters<UpdateGroupRequest, UpdateGroupResponse> {
    return {
        context: "",
        method: "POST",
        path: buildQueryString("/api/hpc/apps" + "/group/update", {}),
        parameters: request,
        reloadId: Math.random(),
        payload: request
    }
}

export function deleteGroup(
    request: DeleteGroupRequest
): APICallParameters<DeleteGroupRequest, DeleteGroupResponse> {
    return {
        context: "",
        method: "DELETE",
        path: buildQueryString("/api/hpc/apps" + "/group", {}),
        parameters: request,
        reloadId: Math.random(),
        payload: request
    }
}

export type AppOrTool = "APPLICATION" | "TOOL";

export interface UploadLogoProps {
    type: AppOrTool;
    file: File;
    name: string;
}

export async function uploadLogo(props: UploadLogoProps): Promise<boolean> {
    const token = await Client.receiveAccessTokenOrRefreshIt();

    return new Promise((resolve) => {
        const request = new XMLHttpRequest();
        const context = props.type === "APPLICATION" ? "apps" : "tools";
        request.open("POST", Client.computeURL("/api", `/hpc/${context}/uploadLogo`));
        request.setRequestHeader("Authorization", `Bearer ${token}`);
        request.responseType = "text";
        request.setRequestHeader("Upload-Name", b64EncodeUnicode(props.name));
        request.onreadystatechange = () => {
            if (request.status !== 0) {
                if (!inSuccessRange(request.status)) {
                    let message = "Logo upload failed";
                    try {
                        message = JSON.parse(request.responseText).why;
                    } catch (e) {
                        // tslint:disable-next-line: no-console
                        console.log(e);
                        // Do nothing
                    }

                    snackbarStore.addFailure(message, false);
                    resolve(false);
                } else {
                    if (props.type === "APPLICATION") appLogoCache.forget(props.name);
                    else toolLogoCache.forget(props.name);
                    resolve(true);
                }
            }
        };

        request.send(props.file);
    });
}

export interface ClearLogoProps {
    type: AppOrTool;
    name: string;
}

export function clearLogo(props: ClearLogoProps): APICallParameters<ClearLogoProps> {
    const context = props.type === "APPLICATION" ? "apps" : "tools";
    return {
        reloadId: Math.random(),
        method: "DELETE",
        path: `/hpc/${context}/clearLogo`,
        parameters: props,
        payload: props
    };
}

export interface UploadDocumentProps {
    type: AppOrTool;
    document: File;
}

export async function uploadDocument(props: UploadDocumentProps): Promise<boolean> {
    const token = await Client.receiveAccessTokenOrRefreshIt();

    return new Promise(resolve => {
        const request = new XMLHttpRequest();
        const context = props.type === "APPLICATION" ? "apps" : "tools";
        request.open("PUT", Client.computeURL("/api", `/hpc/${context}`));
        request.setRequestHeader("Authorization", `Bearer ${token}`);
        request.responseType = "text";
        request.onreadystatechange = () => {
            if (request.status !== 0) {
                if (!inSuccessRange(request.status)) {
                    let message = "Upload failed";
                    try {
                        message = JSON.parse(request.responseText).why;
                    } catch (e) {
                        console.log(e);
                        console.log(request.responseText);
                        // Do nothing
                    }
                    snackbarStore.addFailure(message, false);
                    resolve(false);
                } else {
                    resolve(true);
                }
            }
        };

        request.send(props.document);
    });
}
