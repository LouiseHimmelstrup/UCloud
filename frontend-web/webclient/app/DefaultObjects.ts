import {SidebarOption, Page} from "Types";
import {Status} from "Navigation";
import {Analysis, DetailedApplicationSearchReduxState, RunsSortBy} from "Applications";
import {File, DetailedFileSearchReduxState} from "Files";
import {SortOrder, SortBy} from "Files";
import {DashboardStateProps} from "Dashboard";
import {Publication} from "Zenodo";
import {Notification} from "Notifications";
import {Upload} from "Uploader";
import {Activity, ActivityGroup, ActivityFilter} from "Activity";
import {Reducer} from "redux";
import {SimpleSearchStateProps} from "Search";
import * as ApplicationRedux from "Applications/Redux";
import * as AccountingRedux from "Accounting/Redux";
import * as SnackbarRedux from "Snackbar/Redux";
import * as FavoritesRedux from "Favorites/Redux";
import {defaultAvatar} from "UserSettings/Avataaar";
import {SidebarPages} from "ui-components/Sidebar";
import {ScrollResult} from "Scroll/Types";
import * as ProjectRedux from "Project/Redux";

export const DefaultStatus: Status = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended."
};

export enum KeyCode {
    ENTER = 13,
    ESC = 27
}

export const emptyPage: Page<any> = {items: [], itemsPerPage: 25, itemsInTotal: 0, pageNumber: 0, pagesInTotal: 0};

export enum AnalysesStatusMap {
    "PENDING",
    "IN PROGRESS",
    "COMPLETED"
}

export enum RightsNameMap {
    "NONE" = "None",
    "READ" = "Read",
    "READ_WRITE" = "Read/Write"
}

export enum SensitivityLevel {
    "INHERIT" = "Inherit",
    "PRIVATE" = "Private",
    "CONFIDENTIAL" = "Confidential",
    "SENSITIVE" = "Sensitive"
}

export type Sensitivity = keyof typeof SensitivityLevel;

export enum SensitivityLevelMap {
    INHERIT = "INHERIT",
    PRIVATE = "PRIVATE",
    CONFIDENTIAL = "CONFIDENTIAL",
    SENSITIVE = "SENSITIVE"
}

function getFilesSortingColumnOrDefault(columnIndex: 0 | 1): SortBy {
    const sortingColumn = window.localStorage.getItem(`filesSorting${columnIndex}`);
    if (sortingColumn && Object.values(SortBy).includes(sortingColumn)) return sortingColumn as SortBy;
    switch (columnIndex) {
        case 0:
            window.localStorage.setItem("filesSorting0", SortBy.MODIFIED_AT);
            return SortBy.MODIFIED_AT;
        case 1:
            window.localStorage.setItem("filesSorting1", SortBy.SIZE);
            return SortBy.SIZE;
    }
}

function getItemOrDefault<T, T2>(itemName: string, defaultValue: T, en: T2): T {
    const item = window.localStorage.getItem(itemName);
    if (item && Object.values(en).includes(item)) {
        return item as unknown as T;
    }
    return defaultValue;
}

export interface ComponentWithLoadingState {
    loading: boolean
    error?: string
}

export interface ComponentWithPage<T> extends ComponentWithLoadingState {
    page: Page<T>
}

export interface ComponentWithScroll<Item, OffsetType> extends ComponentWithLoadingState {
    scroll?: ScrollResult<Item, OffsetType>
}

export interface ResponsiveReduxObject {
    mediaType: string
    orientation: string
    lessThan: Record<string, boolean>
    greaterThan: Record<string, boolean>
    is: Record<string, boolean>
}

export const initResponsive = (): ResponsiveReduxObject => ({
    mediaType: "",
    orientation: "",
    lessThan: {},
    greaterThan: {},
    is: {},
});

export interface FilePreviewReduxState {
    file?: File
    error?: string
}

export interface FilesReduxObject extends ComponentWithPage<File> {
    sortOrder: SortOrder
    sortBy: SortBy
    path: string
    filesInfoPath: string
    fileSelectorError?: string
    sortingColumns: [SortBy, SortBy]
    fileSelectorLoading: boolean
    fileSelectorShown: boolean
    fileSelectorPage: Page<File>
    fileSelectorPath: string
    fileSelectorCallback: Function
    fileSelectorIsFavorites: boolean
    disallowedPaths: string[]
    invalidPath: boolean
}

export interface FileInfoReduxObject {
    file?: File
    error?: string
    activity: Page<Activity>
    loading: boolean
}

export interface AnalysisReduxObject extends ComponentWithPage<Analysis> {
    sortBy: RunsSortBy
    sortOrder: SortOrder
}

export interface NotificationsReduxObject {
    redirectTo: string
    items: Notification[]
    loading: boolean
    error?: string
}

export interface ZenodoReduxObject extends ComponentWithPage<Publication> {
    connected: boolean
}

export interface StatusReduxObject {
    status: Status
    title: string
    page: SidebarPages
    loading: boolean
}

export interface SidebarReduxObject {
    pp: boolean
    options: SidebarOption[]
    kcCount: number
}

export interface HeaderSearchReduxObject {
    prioritizedSearch: HeaderSearchType
    refresh?: () => void
}

export interface RunApplicationReduxObject {

}

export type ActivityReduxObject = ComponentWithScroll<ActivityGroup, number> & ActivityFilter;

export type HeaderSearchType = "files" | "applications" | "projects";

export interface UploaderReduxObject {
    uploads: Upload[]
    visible: boolean
    path: string
    allowMultiple: boolean
    onFilesUploaded: (p: string) => void
    error?: string
    loading: boolean
}

interface LegacyReducers {
    dashboard?: Reducer<DashboardStateProps>
    files?: Reducer<FilesReduxObject>
    uploader?: Reducer<UploaderReduxObject>
    status?: Reducer<StatusReduxObject>
    notifications?: Reducer<NotificationsReduxObject>
    analyses?: Reducer<AnalysisReduxObject>
    zenodo?: Reducer<ZenodoReduxObject>
    header?: Reducer<HeaderSearchReduxObject>
    sidebar?: Reducer<SidebarReduxObject>
    activity?: Reducer<ActivityReduxObject>
    detailedResult?: Reducer<DetailedResultReduxObject>
}

export type Reducers = LegacyReducers & ApplicationRedux.Reducers & AccountingRedux.Reducers;

export type DetailedResultReduxObject = ComponentWithPage<File>

export const initDetailedResult = (): DetailedResultReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined
});

/* FIXME */
interface LegacyReduxObject {
    dashboard: DashboardStateProps
    files: FilesReduxObject,
    uploader: UploaderReduxObject
    status: StatusReduxObject,
    notifications: NotificationsReduxObject
    analyses: AnalysisReduxObject
    zenodo: ZenodoReduxObject
    header: HeaderSearchReduxObject
    sidebar: SidebarReduxObject
    activity: ActivityReduxObject
    detailedResult: DetailedResultReduxObject
    simpleSearch: SimpleSearchStateProps
    detailedFileSearch: DetailedFileSearchReduxState
    detailedApplicationSearch: DetailedApplicationSearchReduxState
    fileInfo: FileInfoReduxObject
    avatar: AvatarReduxObject
    filePreview: FilePreviewReduxState
    responsive?: ResponsiveReduxObject
    project: ProjectRedux.State
    loading?: boolean
}

export type ReduxObject =
    LegacyReduxObject &
    ApplicationRedux.Objects &
    AccountingRedux.Objects &
    FavoritesRedux.Objects &
    SnackbarRedux.Wrapper;

export const initActivity = (): ActivityReduxObject => ({
    loading: false
});

export const initNotifications = (): NotificationsReduxObject => ({
    items: [],
    loading: false,
    redirectTo: "",
    error: undefined
});

export const initHeader = (): HeaderSearchReduxObject => ({
    prioritizedSearch: "files"
});

export const initStatus = (): StatusReduxObject => ({
    status: DefaultStatus,
    title: "",
    page: SidebarPages.None,
    loading: false
});

export const initDashboard = (): DashboardStateProps => ({
    favoriteFiles: [],
    recentFiles: [],
    recentAnalyses: [],
    notifications: [],
    favoriteLoading: false,
    recentLoading: false,
    analysesLoading: false
});

export function initObject(homeFolder: string): ReduxObject {
    return {
        dashboard: initDashboard(),
        files: initFiles(homeFolder),
        status: initStatus(),
        header: initHeader(),
        notifications: initNotifications(),
        analyses: initAnalyses(),
        zenodo: initZenodo(),
        sidebar: initSidebar(),
        uploader: initUploads(),
        activity: initActivity(),
        detailedResult: initDetailedResult(),
        simpleSearch: initSimpleSearch(),
        detailedApplicationSearch: initApplicationsAdvancedSearch(),
        detailedFileSearch: initFilesDetailedSearch(),
        fileInfo: initFileInfo(),
        avatar: initAvatar(),
        filePreview: initFilePreview(),
        project: ProjectRedux.initialState,
        ...ApplicationRedux.init(),
        ...AccountingRedux.init(),
        ...FavoritesRedux.init(),
        ...SnackbarRedux.init(),
        responsive: undefined,
    }
}


export const initFilePreview = () => ({
    file: undefined,
    error: undefined
});

export type AvatarReduxObject = typeof defaultAvatar & { error?: string };
export const initAvatar = (): AvatarReduxObject => ({...defaultAvatar, error: undefined});

export const initSimpleSearch = (): SimpleSearchStateProps => ({
    files: emptyPage,
    filesLoading: false,
    applications: emptyPage,
    applicationsLoading: false,
    errors: [],
    search: "",
    applicationSearch: initApplicationsAdvancedSearch(),
    fileSearch: initFilesDetailedSearch()
});

export const initAnalyses = (): AnalysisReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined,
    sortBy: RunsSortBy.createdAt,
    sortOrder: SortOrder.ASCENDING
});

export const initZenodo = (): ZenodoReduxObject => ({
    connected: false,
    loading: false,
    page: emptyPage
});

export const initSidebar = (): SidebarReduxObject => ({
    pp: false,
    kcCount: 0,
    options: []
});

export const initUploads = (): UploaderReduxObject => ({
    path: "",
    uploads: [],
    visible: false,
    allowMultiple: false,
    error: undefined,
    onFilesUploaded: () => null,
    loading: false
});

export const initFileInfo = (): FileInfoReduxObject => ({
    activity: emptyPage,
    loading: false
});

export const initFiles = (homeFolder: string): FilesReduxObject => ({
    page: emptyPage,
    sortOrder: getItemOrDefault("sortOrder", SortOrder.ASCENDING, SortOrder),
    sortBy: getItemOrDefault("sortBy", SortBy.PATH, SortBy),
    loading: false,
    error: undefined,
    path: "",
    filesInfoPath: "",
    sortingColumns: [getFilesSortingColumnOrDefault(0), getFilesSortingColumnOrDefault(1)],
    fileSelectorLoading: false,
    fileSelectorShown: false,
    fileSelectorPage: emptyPage,
    fileSelectorPath: homeFolder,
    fileSelectorIsFavorites: false,
    fileSelectorCallback: () => undefined,
    fileSelectorError: undefined,
    disallowedPaths: [],
    invalidPath: false
});

export const initFilesDetailedSearch = (): DetailedFileSearchReduxState => ({
    hidden: true,
    allowFolders: true,
    allowFiles: true,
    fileName: "",
    extensions: new Set(),
    tags: new Set(),
    sensitivities: new Set(),
    createdBefore: undefined,
    createdAfter: undefined,
    modifiedBefore: undefined,
    modifiedAfter: undefined,
    error: undefined,
    loading: false
});

export const initApplicationsAdvancedSearch = (): DetailedApplicationSearchReduxState => ({
    error: undefined,
    loading: false,
    hidden: true,
    appName: "",
    appVersion: "",
    tags: ""
});
