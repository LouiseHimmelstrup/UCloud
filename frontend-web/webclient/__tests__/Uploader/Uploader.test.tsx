import "jest-styled-components";
import * as React from "react";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router";
import {create} from "react-test-renderer";
import {AnyAction} from "redux";
import {initUploads, SensitivityLevelMap} from "../../app/DefaultObjects";
import {Uploader} from "../../app/Uploader";
import {UploadPolicy} from "../../app/Uploader/api";
import * as UploaderActions from "../../app/Uploader/Redux/UploaderActions";
import uploader from "../../app/Uploader/Redux/UploaderReducer";
import {configureStore} from "../../app/Utilities/ReduxUtilities";


// configure({ adapter: new Adapter() });

describe("Uploader", () => {
    test("Closed Uploader component", () => {
        const store = configureStore({
            uploader: initUploads()
        }, {uploader});
        expect(create(
            <Provider store={store}>
                <MemoryRouter>
                    <Uploader />
                </MemoryRouter>
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });

    // FIXME Tests modal, which requires accessing the portal it is being rendered in?
    test.skip("Open Uploader component", () => {
        const store = configureStore({
            uploader: initUploads()
        }, {uploader});
        store.dispatch(UploaderActions.setUploaderVisible(true, "") as AnyAction);
        expect(create(
            <Provider store={store}>
                <MemoryRouter>
                    <Uploader />
                </MemoryRouter>
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });

    test.skip("Render Uploader component with files", () => {
        const store = configureStore({
            uploader: initUploads()
        }, {uploader});
        store.dispatch(UploaderActions.setUploaderVisible(false, "") as AnyAction);
        store.dispatch(UploaderActions.setUploads([{
            file: new File([], "file"),
            isUploading: false,
            sensitivity: SensitivityLevelMap.PRIVATE,
            resolution: UploadPolicy.REJECT,
            uploadEvents: [],
            progressPercentage: 0,
            extractArchive: false,
            uploadXHR: undefined,
            isPending: false,
            path: "",
            uploadSize: 1
        }]) as AnyAction);
        expect(create(
            <Provider store={store}>
                <MemoryRouter>
                    <Uploader />
                </MemoryRouter>
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });
});