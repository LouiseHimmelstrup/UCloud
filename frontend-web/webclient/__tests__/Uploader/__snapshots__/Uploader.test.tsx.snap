// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader Closed Uploader component 1`] = `null`;
